mule-deployer/src/main/java/org/mule/tools/client/AbstractMuleClient.java

```diff
@@ -284,9 +294,15 @@ public String getBusinessGroupIdByBusinessGroupPath() {
     return currentOrgId;
   }
 
-  private String getBearerToken(Credentials credentials) {
+  private String getBearerToken(AnypointCredential credentials) {
     if (isBlank(bearerToken)) {
-      bearerToken = authenticationServiceClient.getBearerToken(credentials);
+      switch (credentials.credentialType()) {
+        case user:
+          Credentials creds = (Credentials) credentials;
+          bearerToken = authenticationServiceClient.getBearerToken(creds);
```

Comment:
```suggestion
          bearerToken = authenticationServiceClient.getBearerToken(creds);
          break;
```

Reply:
Add the breaks inside each case, if they are not present the next statement is executed and is causing a ClassCastException.

Reply:
Thanks! I have added next commit with this fix.